### PR TITLE
CB-12565: DES deletion and env termination fail if the RG name contains a dot

### PR DESCRIPTION
**Pull request by juhi-09**
CB-12565: DES deletion and env termination fail if the RG name contains a dot

This PR fixes the regex pattern to extract resource group name from resource id, specifically for disk encryption set (DES).
As per Azure documentation, resource group name - Can include alphanumeric, underscore, parentheses, hyphen, period (except at end), and Unicode characters that match the allowed characters. 'RESOURCE_GROUP_NAME' pattern is updated to extract the name correctly.

### CODE REVIEW

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java [hunk 1]
```diff
@@ -139,6 +142,40 @@ private void initActionFailedExceptionConversion() {
                 .thenAnswer(invocation -> new Retry.ActionFailedException(new CloudConnectorException(invocation.getArgument(0, Exception.class))));
     }
 
+    private static Object[][] getDesResourceId() {
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java:145]: As a convention, such methods shall be named so their purpose is clear. If used for a single test method, it makes sense to name them as _testMethod_`DataProvider`. So in this case something like `testDiskEncryptionSetResourceGroupNameExtractionDataProvider` would sound more appropriate.

Note: `DataProvider` is a TestNG terminology; it was the first test library that provided support for parameterized tests, way before JUnit.

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java [hunk 2]
```diff
@@ -139,6 +142,40 @@ private void initActionFailedExceptionConversion() {
...
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910"},
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyUnderscroceResourceGroup---___....)()(0910/providers/" +
+                        "Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyUnderscroceResourceGroup---___....)()(0910"}
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("getDesResourceId")
+    public void testDiskEncryptionSetResourceGroupNameExtraction(String desResourceId, String resourceGroupName) {
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java:170]: - To make it easier to identify the individual test cases of a parameterized test, it is a convention to add some "test case description". This is done with `@ParameterizedTest(name = "{0}")` where the value of the `name` property can contain placeholders referencing method arguments; `{0}` is the first (leftmost) one. See https://github.com/hortonworks/cloudbreak/blob/master/environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceEncryptionParametersTest.java#L79-L81 for an example. Either add a dedicated argument like `testCaseName` there, or simply reference one or more arguments that are used to construct the test case description.
- Again, as a convention, method arguments denoting expected values should be named accordingly. So `resourceGroupName` should be renamed to something like `resourceGroupNameExpected` or `expectedResourceGroupName`, whichever you prefer.

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java [hunk 3]
```diff
@@ -139,6 +142,40 @@ private void initActionFailedExceptionConversion() {
                 .thenAnswer(invocation -> new Retry.ActionFailedException(new CloudConnectorException(invocation.getArgument(0, Exception.class))));
     }
 
+    private static Object[][] getDesResourceId() {
+        return new Object[][]{
+                {"/subscriptions/dummySubscriptionId/resourceGroups/dummyAlphaResourceGroup/providers/" +
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java:147]: Not a big deal but it is a helpful convention to add a comment just before the first `{arguments}` entry listing all the desired test method argument names. So, in this case, something like `// desResourceId resourceGroupName` (or adapted accordingly if adding or renaming arguments).

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java [hunk 4]
```diff
@@ -139,6 +142,40 @@ private void initActionFailedExceptionConversion() {
...
+                        "dummyUnderscroceResourceGroup---___....)()(0910"}
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("getDesResourceId")
+    public void testDiskEncryptionSetResourceGroupNameExtraction(String desResourceId, String resourceGroupName) {
+        String rgName = "";
+        Matcher matcher = underTest.DISK_ENCRYPTION_SET_RESOURCE_GROUP.matcher(desResourceId);
+        if (matcher.matches()) {
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java:173]: We always expect the pattern to match, do not we? So I would add an `assertTrue(matcher.matches())` and eliminate the `if` condition entirely. Moreover, the dummy `rgName = ""` initialization is also unnecessary.

**Review by lajosrodek — COMMENTED** [review]
My 2 cents.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java [hunk 5]
```diff
@@ -34,14 +35,16 @@
 
 @Service
 public class AzureEncryptionResources implements EncryptionResources {
+
+    @VisibleForTesting
+    static final Pattern DISK_ENCRYPTION_SET_RESOURCE_GROUP = Pattern.compile(".*resourceGroups/([^/]+)/providers.*");
```
Comment (doktoric) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java:40]: why visible for testing?

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java [hunk 6]
```diff
@@ -139,6 +143,38 @@ private void initActionFailedExceptionConversion() {
...
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyUnderscoreResourceGroup---___....)()(0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyUnderscoreResourceGroup---___....)()(0910"}
+        };
+    }
+
+    @ParameterizedTest(name = "{0}")
```
Comment (doktoric) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java:170]: the name should show what you are testing with some description. This is just a suggestion

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java (hunk 5 above)
Comment (juhi-09) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java:40]: The change is to correct the pattern, using this pattern in test will ensure that future changes to pattern would be tested appropriately.

**Review by doktoric — APPROVED** [review]

**Comment by doktoric** [comment]
rebase pls

**Pull request merged**